tools/pannopi_report.py

```diff
@@ -0,0 +1,172 @@
+            elif line[0] == "Total length (>= 0 bp)":
+                length = line[1]
+                quast_report["Total_length"] = length
+            elif line[0] == "N50":
+                n50 = line[1]
+                quast_report["N50"] = n50
+            elif line[0] == "# misassemblies":
+                misasemblies = line[1]
+                quast_report["Misassemblies"] = misasemblies
+            elif line[0] == "Genome fraction (%)":
+                genome_fraction = line[1]
+                quast_report["Genome_fraction"] = genome_fraction
+    return quast_report
+
+def prokka_report(prokka_file):
+    prokka_report = {}
+    with open(prokka_file) as fh:
+        for line in fh:
+            line = line.strip().split(": ")
+            if line[0] == "CDS":
+                cds = line[1]
+                prokka_report["CDS"] = cds
+            elif line[0] == "rRNA":
+                rrna = line[1]
+                prokka_report["rRNA"] = rrna
+            elif line[0] == "tRNA":
+                trna = line[1]
+                prokka_report["tRNA"] = trna
+    return prokka_report
+
+
+def fastani_report(fastani_file):
+    fastani_report = {}
+    with open(fastani_file) as fh:
+        for line in fh:
+            line = line.strip().split("\t")
+            fastani_report["FastANI"] = line[-3]
+            return fastani_report
+
+def busco_report(busco_dir):
+    busco_files = glob.glob(os.path.join(busco_dir, "*busco.txt"))
+    busco_report = {}
+    for file in busco_files:
+        with open(file) as fh:
+            for line in fh:
+                if line.startswith("# The lineage dataset is:"):
+                    dataset = "busco_" + line.strip().split()[5]
+                line = line.strip()
+                if line.startswith("C:"):
+                    buscos = line.strip().split("%")[0].split(":")[1]
+        busco_report[dataset] = buscos
+    return busco_report
+
+def eggnog_report(eggnog_file):
+    # eggnog_counter
+    eggnog_report = {"Unique_GOs": 0, "Unique_KEGGs": 0}
+    with open(eggnog_file) as fh:
+        go_list = []
+        kegg_ko_list = []
+        for line in fh:
+            if line.startswith("#"):
+                continue
+            line = line.strip().split("\t")
+            go = line[9]
+            if go != "-":
+                match_gos_list = go.split(",")
+                for match in match_gos_list:
+                    if match not in go_list:
+                        go_list.append(match)
+            kegg_ko = line[11]
+            if kegg_ko != "-":
+                match_kos_list = kegg_ko.split(",")
+                for match in match_kos_list:
+                    if match not in kegg_ko_list:
+                        kegg_ko_list.append(match)
+
+        eggnog_report["Unique_GOs"] += len(go_list)
+        eggnog_report["Unique_KEGGs"] += len(kegg_ko_list)
+    return eggnog_report
+
+
+def abricate_summary(ncbi_file, megares_file, virulence_file, plasmods_file):
```

Comment:
you can use here the same settings dictionary, it will be more easy to read file:

file_basename = os.path.basename(settings[file_name])